src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs

```diff
@@ -6791,8 +7161,48 @@ struct S1
             VerifyParameterSymbol(comp.GetMember<MethodSymbol>("R2.F2").ThisParameter, "scoped in R2 this", RefKind.In, DeclarationScope.RefScoped);
         }
 
-        // PROTOTYPE: Test 'scoped' with extension method 'this'.
-        // PROTOTYPE: Test 'scoped' with 'params'.
+        [Fact]
+        public void ExtensionThisScope()
+        {
+            var source =
+@"ref struct R<T> { }
+static class Extensions
+{
+    static void F1(this scoped R<object> r) { }
+    static void F2<T>(scoped this R<T> r) { }
+    static void F3<T>(this scoped ref T t) where T : struct { }
+    static void F4(this ref scoped R<int> r) { }
```

Comment:
consider also checking a `this` parameter with no scoped modifiers as a baseline.